src/worker.c

```diff
@@ -77,7 +78,13 @@ static JSValue worker_eval(JSContext *ctx, int argc, JSValue *argv) {
         goto error;
     }
 
-    ret = TJS_EvalModule(ctx, filename, false);
+    if (!JS_IsUndefined(argv[1])) {
+        const char *blob = JS_ToCString(ctx, argv[1]);
```

Comment:
Use `JS_ToCStringLen` so you get the length straight and can avoid doing `strlen`.